File: backend/app/schemas/withdrawal.py
from sqlalchemy import Column, Integer, Float, DateTime, ForeignKey, Enum, String
from sqlalchemy.orm import relationship
from .base import Base
from .enums import PaymentMethod, WithdrawalStatus

class Withdrawal(Base):
    __tablename__ = "withdrawal"


    withdrawal_id = Column(Integer, primary_key=True, autoincrement=True)
    freelancer_id = Column(Integer, ForeignKey("freelancer.freelancer_id"), nullable=False, index=True)
    admin_id = Column(Integer, ForeignKey("admin.admin_id"), nullable=True, index=True)
    payment_method = Column(String, nullable=False)
    amount = Column(Float, nullable=False)
    requested_at = Column(DateTime, nullable=False)
    withdrawal_status = Column(Enum(WithdrawalStatus), nullable=False)
    processed_at = Column(DateTime, nullable=True)
    paypal_link = Column(String, nullable=True)
    payoneer_email = Column(String, nullable=True)
    wavepay_phone = Column(String, nullable=True)
    kpay_phone = Column(String, nullable=True)
    account_holder_name = Column(String, nullable=True)
    bank_name = Column(String, nullable=True)
    account_number = Column(String, nullable=True)

    freelancer = relationship("Freelancer", back_populates="withdrawals")
    admin = relationship("Admin", back_populates="withdrawals")

File: src/types/withdrawal.ts
export interface Withdrawal {
  freelancer_id: number;
  withdrawal_id: number;
  admin_id: number | null;
  payment_method: string;
  amount: number;
  requested_at: string;
  processed_at: string | null;
  withdrawal_status: string;
  paypal_link: string | null;
  payoneer_email: string | null;
  wavepay_phone: string | null;
  kpay_phone: string | null;
  account_holder_name: string | null;
  bank_name: string | null;
  account_number: string | null;
  proof_of_payment: string | null;
}

export interface WithdrawalRequest {
  freelancer_id: number;
  withdrawal_id: number;
  amount: number;
  requested_at: string;
  processed_at: string | null;
  payoneer_email: string | null;
  kpay_phone: string | null;
  bank_name: string | null;
  admin_id: number | null;
  payment_method: string;
  withdrawal_status: string;
  paypal_link: string | null;
  wavepay_phone: string | null;
  account_holder_name: string | null;
  account_number: string | null;
  proof_of_payment: string | null
}

export interface BalanceResponse {
  current_balance: number;
}

export interface PaymentRequestBody {
  freelancer_id: number;
  payment_method: string;
  amount: number;
  paypal_link?: string;
  payoneer_email?: string;
  wavepay_phone?: string;
  account_holder_name?: string;
  kpay_phone?: string;
  bank_name?: string;
  account_number?: string;
}